### PR TITLE
Add tf cluster status

### PR DESCRIPTION
**Pull request by ScorpioCPH**
Hi, this PR add `TFClusterStatus` to represent all of the tasks' status in the TF distributed cluster.

@gaocegege-caicloud @DjangoPeng @mqliang @ddysher PTAL.

### CODE REVIEW

File: apis/kubeflow/v1alpha1/types.go [hunk 1]
```diff
@@ -88,14 +88,29 @@ type ChiefSpec struct {
...
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Represents the latest available observations of a TFJob object's current state.
 	Conditions []*TFJobCondition `json:"conditions"`
 
+	// TFClusterStatus represents all of the tasks' status in the TF distributed cluster.
+	TFClusterStatus TFClusterStatus `json:"tfClusterStatus"`
+
 	// ReplicaStatuses specify the status of each TFReplica.
 	TFReplicaStatuses []*TFReplicaStatus `json:"tfReplicaStatuses"`
```
Comment (gaocegege) [inline apis/kubeflow/v1alpha1/types.go]: What are the relationship between TFClusterStatus and TFReplicaStatuses?

Comment (ScorpioCPH) [reply apis/kubeflow/v1alpha1/types.go]: There is no relationship, `TFReplicaStatuses` maybe deprecated after discussion with community.

Comment (gaocegege) [reply apis/kubeflow/v1alpha1/types.go]: SGTM

**Comment by gaocegege** [comment]
FYI, the clientset is not updated :-)

**Comment by ScorpioCPH** [comment]
@gaocegege Updated.

**Pull request merged**